src/metrics-logger.ts

```diff
@@ -0,0 +1,77 @@
+'use strict';
+import { Request } from './custom-typings';
+import { Route, RouteSpec, Response, Server } from 'restify';
+import { StatsD, StatsDOptions } from './statsd';
+
+/**
+ * tuple of [seconds, nanoseconds]
+ */
+type HighResolutionTime = [number, number];
+
+export const DEFAULT_KEY_NAME: string = 'handler-0';
+
+export function register(server: Server, statsDOptions: StatsDOptions) {
+    let statsD: StatsD = new StatsD(statsDOptions);
+    let logger = new MetricsLogFactory(statsD).createLogger();
+
+    server.on('after', (request: Request, response: Response, route: Route, error) => {
+        logger(request, response, route);
+    });
+}
+
+export class MetricsLogFactory {
+    constructor(private statsD: StatsD) {
+    }
+
+    createLogger() {
+        return (request: Request, response: Response, route: Route) => {
+            // Ignore requests without timers, e.g. OPTIONS
+            if (!request.timers) {
+                return;
+            }
+
+            let timer = request.timers.find((item) => {
+                return item.name === route.name || item.name === DEFAULT_KEY_NAME;
+            });
+            if (!timer) {
+                return;
+            }
```

Comment:
again, I think if we don't find the timer, we should fall back to the request time method. please create an issue for this and we'll address it at a different time